app/assets/javascripts/application.js

```diff
@@ -12,16 +12,21 @@
 //= require angular-ui-bootstrap
 //= require angular-ui-bootstrap-tpls
 //= require angular-sanitize
+//= require angular-ui-router
+//= require angular-rails-templates
 //= require moment
 //= require moment-strftime/build/moment-strftime.min
 //= require moment-timezone
 //= require sprintf
 //= require numeral
 //= require miq_api
 //= require miq_angular_application
+//= require miq_angular_configuration
+//= require_tree ./components/
+//= require_tree ./controllers/
 //= require_tree ./directives/
+//= require_tree ./factories/
```

Comment:
It's because forward compatibility .. if you have a `service`, you won't have to rewrite it for angular 2 (well, at least not this aspect ;)), no factories there..

But you're right that SSUI does prefer to use factories, probably mostly because john papa's angular guide says so..

But really, I'd say, if it looks like a class with some methods on it, `service` is the right tool for the job. If you want to inject a constructor, or a function or something instead, factory makes more sense.. :)

As for that commit.. it *should* work .. but the canonical way to do that would be to not return the object but assign stuff to `this` instead..